gpkit/nomials/variables.py

```diff
@@ -97,20 +108,25 @@ class ArrayVariable(NomialArray):
     """
 
     def __new__(cls, shape, *args, **descr):
-        # pylint: disable=too-many-branches
+        # pylint: disable=too-many-branches, too-many-statements
         cls = NomialArray
 
         if "idx" in descr:
             raise KeyError("the description field 'idx' is reserved")
 
         shape = (shape,) if isinstance(shape, Numbers) else tuple(shape)
+        from .. import VECTORIZATION
+        if VECTORIZATION:
+            shape = shape + tuple(VECTORIZATION)
+            descr["vectorization"] = VECTORIZATION
 
         descr["shape"] = shape
 
         for arg in args:
             if isinstance(arg, Strings) and "name" not in descr:
                 descr["name"] = arg
-            elif (isinstance(arg, Iterable) and not isinstance(arg, Strings)
+            elif (isinstance(arg, (Numbers, Iterable))
+                  and not isinstance(arg, Strings)
                   and "value" not in descr):
                 descr["value"] = arg
             elif isinstance(arg, Strings+(Quantity,)) and "units" not in descr:
```

Comment:
Yeah, except for the `values` bit they're the same. It's pretty short, though; we wouldn't save lines of code by abstracting it out. Do you think it would read more clearly?